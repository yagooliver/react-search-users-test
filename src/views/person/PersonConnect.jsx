import {connect} from 'react-redux';
import PersonContainer from './components/PersonContainer';
import {
  loadPersons,
  changeQuery,
  loadQueries,
  changePage,
} from './action/actions';

const mapStateToProps = (state) => ({
  persons: state.person.persons,
  totalItems: state.person.totalItems,
  totalPage: state.person.totalPage,
  page: state.person.page,
  size: state.person.size,
  query: state.person.query,
  isFetching: state.person.isFetching
});

const mapDispatchToProps = (dispatch) => ({
  loadPersons: (page, size) => dispatch(loadPersons(page, size)),
  changeQuery: (item) => dispatch(changeQuery(item)),
  loadQueries: (query) => dispatch(loadQueries(query)),
  changePage: (currentPage) => dispatch(changePage(currentPage))
});

export default connect(mapStateToProps, mapDispatchToProps)(PersonContainer);