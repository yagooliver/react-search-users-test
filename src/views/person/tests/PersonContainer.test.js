import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import PersonConnect from '../PersonConnect';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let store;
const initialState = {
  person: {
    persons: [],
    firstState: [],
    allItems: [],
    totalItems: 0,
    totalPage: 0,
    page: 0,
    size: 10,
    query: {
      name: '',
      age: '',
    },
    isFetching: false
  }
}
store = mockStore(initialState);
configure({adapter: new Adapter()});

describe("When screen is loaded", () => {
  let wrapper;
  it("should show title", () => {
    store = mockStore(initialState);
    wrapper = shallow(<PersonConnect store={store}/>);
    expect(wrapper.render(/Title/).length).toBe(1);
  });
    
  it("should show load when list is empty", () => {
    store = mockStore(initialState);
    wrapper = shallow(<PersonConnect store={store}/>)
    expect(wrapper.find("LoadGif").exists()).toMatchSnapshot();
  });

  it("shouldn't show loading when list is not empty", () => {
    store = mockStore(initialState);
    store.dispatch({
      type: 'PERSONS_LOAD_PERSONS',
      payload: {
        data: [{
          name: 'Test 1',
          age: 20
        }, {
          name: 'Test 1',
          age: 20
        }], 
        total: 2, 
        totalPages: 1,
        allItems: [{
          name: 'Test 1',
          age: 20
        }, {
          name: 'Test 1',
          age: 20
        }]
      }
    });
    wrapper = shallow(<PersonConnect store={store}/>)
    expect(wrapper.find("PersonListComponent").exists()).toMatchSnapshot();
  })
});
