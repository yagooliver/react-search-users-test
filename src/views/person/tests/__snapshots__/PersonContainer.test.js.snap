// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When screen is loaded should show load when list is empty 1`] = `false`;

exports[`When screen is loaded shouldn't show loading when list is not empty 1`] = `false`;
